File: test-scripts/lua/default-hello.lua
local gopherbot = require "gopherbot_v1"

local task = gopherbot.task
local Robot = gopherbot.Robot

local default_config = [[
---
Commands:
  - Regex: (?i:lua default hello)
    Command: hello
]]

local cmd = arg and arg[1] or ""

if cmd == "_configure" then
  return default_config
end

if cmd == "_init" then
  return task.Normal
end

local bot = Robot:new()

if cmd == "hello" then
  local bot_name = bot:GetBotAttribute("fullName")
  local sender = bot:GetSenderAttribute("fullName")
  bot:Say("Lua says hello from " .. tostring(bot_name) .. " to " .. tostring(sender) .. " in #" .. tostring(bot.channel) .. ".")
  bot:Reply("This script only needs the installed default fixture.")
  return task.Normal
end

bot:Log(gopherbot.log.Error, "unknown Lua default command: " .. tostring(cmd))
return task.Fail

File: test-scripts/javascript/default-hello.js
const { Robot, task, log } = require('gopherbot_v1')();

const defaultConfig = `---
Commands:
  - Regex: (?i:js default hello)
    Command: hello
`;

function main(argv) {
  const cmd = argv[2] || "";
  if (cmd === "_configure") {
    return defaultConfig;
  }
  if (cmd === "_init") {
    return task.Normal;
  }

  const bot = new Robot();
  if (cmd === "hello") {
    const botName = bot.GetBotAttribute("fullName").attribute || "the robot";
    const sender = bot.GetSenderAttribute("fullName").attribute || bot.user;
    bot.Say(`JavaScript says hello from ${botName} to ${sender} in #${bot.channel}.`);
    bot.Reply("This script only needs the installed default fixture.");
    return task.Normal;
  }

  bot.Log(log.Error, `unknown JavaScript default command: ${cmd}`);
  return task.Fail;
}

main(process.argv || []);
